### PR TITLE
[4.0] Correcting frontend login/registration menu items password label

### PR DESCRIPTION
**Pull request by infograf768**
# Summary of Changes
The label of the Password field is marked as **(optional)** although it is required.
Modified the renderfied.php layout

### Testing Instructions
Create a Login Form menu item and a Registration form menu item
Display in frontend

### Before patch
login_before

### After patch
login_after

### CODE REVIEW

File: components/com_users/layouts/joomla/form/renderfield.php [hunk 1]
```diff
@@ -43,7 +43,8 @@
  */
 preg_match('/class=\"([^\"]+)\"/i', $input, $match);
 
-$required     = (strpos($input, 'aria-required="true"') !== false || (!empty($match[1]) && strpos($match[1], 'required') !== false));
+$required     = (strpos($input, 'aria-required="true"') !== false
+				|| (!empty($match[1]) && (strpos($match[1], 'required') !== false) || strpos($displayData['class'], 'required') !== false));
```
Comment (Quy) [inline components/com_users/layouts/joomla/form/renderfield.php]:
```suggestion
			|| (!empty($match[1]) && (strpos($match[1], 'required') !== false) || strpos($displayData['class'], 'required') !== false));
```

**Comment by brianteeman** [comment]
Why do we even have this layout override?

**Comment by infograf768** [comment]
See https://github.com/joomla/joomla-cms/pull/18146 concerning ReCaptcha

The reason we have this issue in 4.0 is that was added a new div with `class="password-group"` and `$match[1]` is not anymore the $input id class but that div class.

The new div is added in `ROOT/layouts/joomla/form/field/password.php`

**Comment by richard67** [comment]
I have tested this item :white_check_mark: successfully on b2e7e5b76147819bbc13fa3851bba282f401f575<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/25533">issues.joomla.org/tracker/joomla-cms/25533</a>.</sub>

**Comment by alikon** [comment]
I have tested this item :white_check_mark: successfully on b2e7e5b76147819bbc13fa3851bba282f401f575<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/25533">issues.joomla.org/tracker/joomla-cms/25533</a>.</sub>

**Comment by ghost** [comment]
Status "Ready To Commit".

**Comment by wilsonge** [comment]
Thanks!

**Pull request merged**